File: 2Python应用/f数据编码和处理/6.f
#解析和修改XML
#修改一个XML文档容易，但是要注意所有的修改都是针对父节点元素，将它作为一个
#列表来处理。 remove() insert() append()  element[i] element[i:j]

>>>from xml.etree.ElementTree import parse, Element
>>>doc = parse('pred.xml')
>>>root = doc.getroot()

>>>#Remove a few elements
>>>root.remove(root.find('sri'))
>>>root.remove(root.find('cr'))

>>>#Insert a new element after <nm>...</nm>
>>>root.getchildren().index(root.find('nm'))
1
>>>e = Element('spam')
>>>e.text = 'This is a test'
>>>root.insert(2,e)

>>>#Write back to a file
>>>doc.write('newpred.xml', xml_declaration = True)

File: 2Python应用/f数据编码和处理/6.c
#解析简单的XML数据
#可以使用xml.etree.ElementTree模块从简单的XML文档中提取数据
#xml.etree.ElementTree.parse()函数解析整个XML文档并将其转换成一个文档对象。
#然后，就能使用find() iterfind() findtext()等方法来搜索特定的XML元素。
#ElementTress模块中的每个元素有一些重要的属性和方法，在解析的时候很有用。
#tag属性包含了标签的名字，text属性包含内部文本，get()方法能获取属性值。
#对于更高级的应用程序，可以考虑使用lxml。
from urllib.request import urlopen
from xml.etree.ElementTree import parse

#Download the RSS feed and parse it
u = urlopen('http://planet.python.org/rss20.xml')
doc = parse(u)

#Extract and output tags of interest
for item in doc.iterfind('channel/item'):
	title = item.findtext('title')
	date = item.findtext('pubData')
	link = item.findtext('link')

	print(title)
	print(date)
	print(link)
	print()


>>> doc
<xml.etree.ElementTree.ElementTree object at 0x101339510>
>>> e = doc.find('channel/title')
>>> e
<Element 'title' at 0x10135b310>
>>> e.tag
'title'
>>> e.text
'Planet Python'
>>> e.get('some_attribute')
